caret_analyze/plot/bokeh/callback_info_interface.py

```diff
@@ -101,54 +100,152 @@ def _get_fig_args(self, x_axis_label: str, y_axis_label: str, ywheel_zoom: bool)
         return fig_args
 
     def _show_with_index(self, ywheel_zoom: bool):
+        Hover = HoverTool(
+                tooltips="""
+                <div style="width:400px; word-wrap: break-word;">
+                <br>
+                node_name = @node_name <br>
+                callback_type = @callback_type <br>
+                @callback_param <br>
+                symbol = @symbol
+                </div>
+                """,
+                point_policy='follow_mouse'
+                )
         source_df = self._to_dataframe_core('index')
         l1_columns = source_df.columns.get_level_values(1).to_list()
-        colors = d3['Category20'][20]
         fig_args = self._get_fig_args('index', l1_columns[1], ywheel_zoom)
+        frame_min, frame_max = get_range(self._callbacks)
         p = figure(**fig_args)
-
-        for i, callback_name in enumerate(source_df.columns.get_level_values(0).to_list()):
-            single_cb_df = source_df.loc[:, (callback_name,)].dropna()
-            p.line(x=single_cb_df.index,
-                   y=single_cb_df.loc[:, l1_columns[1]].to_list(),
-                   line_color=colors[i],
-                   legend_label=callback_name)
-
+        p.add_tools(Hover)
+        coloring_rule = 'callback'
+        color_selector = ColorSelector.create_instance(coloring_rule)
+        for i, callback in enumerate(self._callbacks):
+            color = color_selector.get_color(
+                callback.node_name,
+                None,
+                callback.callback_name)
+            line_source = get_callback_lines(callback, source_df, l1_columns, frame_min, 'index')
+            p.line('x',
+                   'y',
+                   source=line_source,
+                   legend_label=f'callback{i}',
+                   color=color)
         p.add_layout(p.legend[0], 'right')
         show(p)
 
     def _show_with_sim_time(self, ywheel_zoom: bool):
+        Hover = HoverTool(
```

Comment:
The show_with_** function seems to do almost the same procedure.
Could you please refactor it?